packages/dev/core/src/AudioV2/abstractAudio/spatialAttachers/spatialAudioAttacher.ts

```diff
@@ -0,0 +1,205 @@
+import type { Camera } from "../../../Cameras/camera";
+import type { AbstractMesh, TransformNode } from "../../../Meshes";
+import type { Nullable } from "../../../types";
+import type { _AbstractSpatialAudioAttacher, ISpatialAudioNode } from "./abstractSpatialAudioAttacher";
+import { _CreateSpatialAudioCameraAttacherAsync } from "./spatialAudioCameraAttacher";
+import { _CreateSpatialAudioMeshAttacherAsync } from "./spatialAudioMeshAttacher";
+import { _CreateSpatialAudioTransformNodeAttacherAsync } from "./spatialAudioTransformNodeAttacher";
+
+export const _SpatialAudioAttachedEntity = {
+    CAMERA: "Camera",
+    MESH: "Mesh",
+    TRANSFORM_NODE: "TransformNode",
+};
+
+export const enum SpatialAudioAttachmentType {
+    POSITION = 1,
+    ROTATION = 2,
+    POSITION_AND_ROTATION = 3,
+}
+
+/**
+ * Provides a common interface for attaching an audio listener or source to a specific entity, ensuring only one entity
+ * is attached at a time.
+ */
+export class _SpatialAudioAttacher {
+    private _attachedEntity: Nullable<AbstractMesh | Camera | TransformNode> = null;
+    private _attacher: Nullable<_AbstractSpatialAudioAttacher> = null;
+    private _attachmentType: SpatialAudioAttachmentType;
+    private _isReadyPromise: Nullable<Promise<void>> = null;
+    private _minUpdateTime: number = 0;
+    private _spatialAudioNode: ISpatialAudioNode;
+
+    /**
+     * Creates a new ExclusiveSpatialAudioAttacher.
+     * @param spatialAudioNode - The spatial audio node to attach to
+     * @param attachmentType - The type of attachment to use; position, rotation or both. Defaults to both
+     */
+    public constructor(spatialAudioNode: ISpatialAudioNode, attachmentType: SpatialAudioAttachmentType = SpatialAudioAttachmentType.POSITION_AND_ROTATION) {
+        this._attachmentType = attachmentType;
+        this._spatialAudioNode = spatialAudioNode;
+    }
+
+    /**
+     * The scene that the audio listener or source is attached to, or null if the audio listener or source is not
+     * attached to a scene.
+     */
+    public get attachedCamera(): Nullable<Camera> {
+        return this._attacher?.getClassName() === _SpatialAudioAttachedEntity.CAMERA ? (this._attachedEntity as Camera) : null;
```

Comment:
how does it work for ArcRotateCamera for instance?

Reply:
Oh, I see what you mean. Thanks for clarifying. The class name being referenced here is for the `_attacher` helper, not the camera.